app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -261,18 +324,33 @@ class AgendaViewModel @Inject constructor(
                 AgendaItem.Event(
                     id = UUID.randomUUID().toString(),
                     title = "Meeting with Jack",
-                    from = ZonedDateTime.of(todayDate, today.toLocalTime().plusHours(5), zoneId),
-                    to = ZonedDateTime.of(todayDate, today.toLocalTime().plusHours(6), zoneId),
-                    remindAt = ZonedDateTime.of(
+                    from = ZonedDateTime.of(
+                        todayDate,
+                        today.toLocalTime().plusHours(3),
+                        zoneId
+                    ),
+                    to = ZonedDateTime.of(
                         todayDate,
                         today.toLocalTime().plusHours(4),
                         zoneId
                     ),
+                    remindAt = ZonedDateTime.of(
+                        todayDate,
+                        today.toLocalTime().plusHours(2),
+                        zoneId
+                    ),
                     description = "Discuss the yet another project",
                     host = "Jack",
                     isUserEventCreator = true,
                     isGoing = false,
-                    attendees = emptyList(),
+                    attendees = listOf(
+                        Attendee(
+                            id = authRepository.getAuthInfo()?.userId!!,
+                            fullName = authRepository.getAuthInfo()?.username ?: "",
+                            email = authRepository.getAuthInfo()?.email ?: "",
```

Comment:
Will this read from preferences 3 times here?

Reply:
The preference is in RAM at this time, so the access is very fast. 

Only when writing it it is slow.

Reply:
> I commented twice on something regarding your events in the VM, but now after reviewing I understand what you're doing. I'm still wondering what's the benefit of sending an event which the VM listens to to update the state instead of just calling a function which updates the state?

I just wanted a centralized place where all the events are processed and updated, and not a bunch of separate functions. I'm mimickying the MVI pattern a bit here. 

## Follow up question

 Is this a bad practice?

Reply:
If that object was already read from memory and just sits in a local variable, your approach is perfectly fine. If new DataStore reads are happening on every call, I'd simply save it in one 👍 

2. question:
Hm, not sure I'd call it a bad practice, it just doesn't feel intuitive to me to go that extra mile if you can just update the state right away using a normal function that expresses the same as your event name. Then you don't need any extra collectors etc.